cut-n-paste/toolbar-editor/egg-toolbars-model.c

```diff
@@ -597,7 +597,8 @@ egg_toolbars_model_load_toolbars (EggToolbarsModel *model,
     return FALSE;
   }
   root = xmlDocGetRootElement (doc);
-
+  if (root == NULL)
+    g_error("Failed to get root element");
```

Comment:
Same case here, the function returns TRUE on success, for this reason you can display a warning message and return FALSE, like `doc == NULL` assert:
```C
   if (doc == NULL)
  {
    g_warning ("Failed to load XML data from %s", xml_file);
    return FALSE;
  }
```
I haven't checked it, but I think it's difficult for the root to be NULL when the document has been parsed correctly.

Reply:
I'm not sure too, but here is am [example](http://xmlsoft.org/tutorial/apc.html) provided by libxml, Here is similar to the example. Maybe in some cases that happens(eg. the document is empty) , adding a check to avoid unexpected errors maybe a good choice.

Reply:
Good find!